app/assets/javascripts/application.js

```diff
@@ -962,4 +962,13 @@ window.clearGeneSearchLoading = function() {
     console.log('Clearing global gene search message');
     $('#wrap').data('spinner').stop();
     $('#gene-search-results-count').html($('.gene-panel').length);
-};
\ No newline at end of file
+};
+
+// force login on ajax 401
+$(document).ajaxError(function (e, xhr, settings) {
+    if (xhr.status == 401) {
```

Comment:
Strict equality would be a bit better:

```suggestion
    if (xhr.status === 401) {
```